### DISCUSSION TITLE
Allow >768x768 images with SD1.4 & SD1.5

### DISCUSSION THREAD

**Discussion by dathide** [NVIDIA/Stable-Diffusion-WebUI-TensorRT#28] [Ideas]
Thanks for your work on this! It's so much faster. However, all of my SD1.5 workflows these days involve generating images with higher resolutions than 768x768. My current favorite starting resolution is 512x1024. I tried editing models.py and ui_trt.py to allow up to 1024x1024 images, but I failed. I know there would be costs for higher resolution support, but I think my 3090 can handle it. I would love a simple way to edit the min and max resolution values for the default engine.

Edit: Something like adding a SD1.5 "Portrait" option for 512x768 to 768x1024 and a SD1.5 "Landscape" option for 768x512 to 1024x768.

**Comment by contentis** [comment]
This should be supported by simply exporting a model with the desired settings. I just realized it is not super obvious, but here is how to do it:

1. Go to the TensorRT tab 
2. Select any of the presets from the dropdown
   * An "Advanced Settings" accordion should appear
3. Now you can modify all values and export any kind of engine. If 512x1024 is your preferred settings it might make sense to export an static engine with that optimal profile
<img width="722" alt="Screenshot 2023-10-18 at 11 10 17" src="https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/29352092/f22a0e4d-0b9e-400d-a2e0-66d6eb0ca61d">

If you only need that specific resolution, check the "Static Engine" checkbox. This will limit the engine to that single config but should compile faster and guarantee best performance. 
<img width="729" alt="Screenshot 2023-10-18 at 11 10 45" src="https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/29352092/0cb7f6ea-a085-41ed-9224-37e693a005f8">

**Comment by tyen0** [comment]
Note the prompt limitations.  I was getting obscure "all NaN" errors until realizing it was due to my prompt being larger than 75 tokens and creating a static shape doesn't let you choose different min and max for tokens in addition to the expected resolution limit.

**Comment by seedhunterai** [comment]
@contentis  Rather than open a new thread related this is working for me at 1920 x 1024 on an a6000..so far 

@dathide this prompt appears to work well for all after you find a seed that doesn't double _((force composition to portrait))_ the other way is put the scene in a room with prompts

Height | 768 | 1024 | 1024
Width | 768 | 1920 | 1920
Batch Size | 1 | 1 | 4
Text-length | 75 | 75 | 225

this is what it looks like running. I've figured out how to also use beyond the prompt values also no idea how but I've been using it awhile over limits in 1.5

![Screenshot_2024_01_09-3](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/156172003/9d1623e3-3eff-46db-af6f-20c40ea928de)
![Screenshot_2024_01_09-4](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/156172003/9b04387d-be16-47de-b7d4-a8f4ca1f2250)

here is an example of output 150 steps to show it actually works if you know what to draw and how to use prompts to work around models trained on lower resolutions. The key is in seed hunting for the seed that will not display a double image i have now 600k images under belt
![Screenshot_2024_01_10-1](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/156172003/b740eb34-00c9-49d7-a801-e202c09d730d)
